Source/Core/AI/ThinkNodes/ThinkNode_Labor.cs

```diff
@@ -1,59 +1,308 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using PrisonLabor.Core.LaborArea;
+using PrisonLabor.Core.LaborWorkSettings;
+using PrisonLabor.Core.Meta;
 using PrisonLabor.Core.Needs;
 using PrisonLabor.Core.Other;
-using PrisonLabor.Core.Trackers;
 using RimWorld;
+using UnityEngine.Assertions.Must;
 using Verse;
 using Verse.AI;
 
 namespace PrisonLabor.Core.AI.ThinkNodes
 {
-    internal class ThinkNode_Labor : ThinkNode_Conditional
+    public class ThinkNode_Labor : ThinkNode
     {
-        protected override bool Satisfied(Pawn pawn)
+        public bool emergency;
+
+        public object Tutorials { get; private set; }
+
+        private readonly int jobsPerTick = 7;
+
+        private static TickManager manager = null;
+
+        private static Dictionary<Pawn, int> lastIndex = new Dictionary<Pawn, int>();
+
+        public override ThinkResult TryIssueJobPackage(Pawn pawn, JobIssueParams jobParams)
         {
-            if (pawn.IsPrisoner)
+            var mneed = pawn.needs.TryGetNeed<Need_Motivation>();
+
+            // Check for medical needs
+            if (HasMedicalNeed(pawn))
+                return SetNotWorking(mneed);
+
+            // Check the timetables in the worktab 
+            if (!HasWorkTimeAssignment(pawn))
+                return SetNotWorking(mneed);
+
+            // Check if the motivation is enabled
+            if (PrisonLaborPrefs.EnableMotivationMechanics && (mneed == null || mneed.IsLazy))
+                return SetNotWorking(mneed);
+
+            // Setting the tickmanager
+            if (manager == null) { manager = Find.TickManager; }
+
+            // TODO Remove this..
+            // Work prisoners will do]
+            if (pawn.timetable == null)
             {
-                //show tutorials
-                Tutorials.Introduction();
-                Tutorials.Management();
+                WorkSettings.InitWorkSettings(pawn);
+
+                // TODO: improve locks
+                // Restrict pawn to labor area
+                pawn.playerSettings.AreaRestriction = pawn.Map.areaManager.Get<Area_Labor>();
```

Comment:
Prisoners should not be restricted to labor arena

Reply:
Well save some time when pathfinding

Reply:
The overhaul for the escape mechanics should account for this. (this build only goal is performance so the mainstream can overtime have a more optimized design not vise versa)